contracts/contracts/token/OUSD.sol

```diff
@@ -75,91 +89,92 @@ contract OUSD is Initializable, InitializableERC20Detailed, Governable {
+     * @return High resolution total number of rebasing credits
      */
-    function rebasingCreditsPerTokenHighres() public view returns (uint256) {
-        return _rebasingCreditsPerToken;
+    function rebasingCreditsHighres() external view returns (uint256) {
+        return rebasingCredits_;
     }
 
     /**
-     * @return High resolution total number of rebasing credits
+     * @return Low resolution total number of rebasing credits
      */
-    function rebasingCreditsHighres() public view returns (uint256) {
-        return _rebasingCredits;
+    function rebasingCredits() external view returns (uint256) {
+        return rebasingCredits_ / RESOLUTION_INCREASE;
     }
 
     /**
-     * @dev Gets the balance of the specified address.
+     * @notice Gets the balance of the specified address.
      * @param _account Address to query the balance of.
      * @return A uint256 representing the amount of base units owned by the
      *         specified address.
      */
-    function balanceOf(address _account)
-        public
-        view
-        override
-        returns (uint256)
-    {
-        if (_creditBalances[_account] == 0) return 0;
-        return
-            _creditBalances[_account].divPrecisely(_creditsPerToken(_account));
+    function balanceOf(address _account) public view returns (uint256) {
+        RebaseOptions state = rebaseState[_account];
+        if (state == RebaseOptions.YieldDelegationSource) {
+            // Saves a slot read when transferring to or from a yield delegating source
+            // since we know creditBalances equals the balance.
+            return creditBalances[_account];
+        }
+        uint256 baseBalance = (creditBalances[_account] * 1e18) /
+            _creditsPerToken(_account);
+        if (state == RebaseOptions.YieldDelegationTarget) {
+            // creditBalances of yieldFrom accounts equals token balances
+            return baseBalance - creditBalances[yieldFrom[_account]];
+        }
+        return baseBalance;
     }
 
     /**
-     * @dev Gets the credits balance of the specified address.
+     * @notice Gets the credits balance of the specified address.
      * @dev Backwards compatible with old low res credits per token.
      * @param _account The address to query the balance of.
      * @return (uint256, uint256) Credit balance and credits per token of the
      *         address
      */
     function creditsBalanceOf(address _account)
-        public
+        external
         view
         returns (uint256, uint256)
     {
         uint256 cpt = _creditsPerToken(_account);
-        if (cpt == 1e27) {
-            // For a period before the resolution upgrade, we created all new
-            // contract accounts at high resolution. Since they are not changing
-            // as a result of this upgrade, we will return their true values
-            return (_creditBalances[_account], cpt);
+        if (cpt == 1e27 || cpt == 1e18) {
```

Comment:
Discontinuities really scare me. 1e27 is a value that we will never see again. But 1e18 is a value below the currently moving value. It could theoretically happen that suddenly almost all accounts in the system report 1e9 times more credits than they had before.

This `creditsBalanceOf` is depreciated, and only for backwards compatibility. Maybe we should note that more.

Reply:
hmm yes this is a great point. The global _creditsPerToken will cross the 1e18 value. I will revert the function to the previous implementation. 

Note: just checked there is no natspect @deprecated tag yet.